### DISCUSSION TITLE
How to check disabled user in external IdP?

### DISCUSSION THREAD

**Discussion by nbasov** [zitadel/zitadel#5914] [General]
Hello! 

Trying Zitadel as SSO with connected external IdP (OKTA by Generic OIDC). I am using test web application from https://github.com/zitadel/react-user-authentication with linked `User Agent` application as described in https://zitadel.com/docs/examples/login/react. Authentications works great with active OKTA users and I got `access_token` and `refresh_token`! 

Next I disable user in OKTA side. I still able to create new Zitadel `refresh_token` (base on `refresh_token` I have got before) using https://zitadel.com/docs/apis/openidoauth/endpoints#token_endpoint Zitadel API.

So, I have several questions:

1) How can I check user state (enabled/disabled) in external IdP using Zitadel API? 
2) Is it possible to get notification from Zitadel about user state changes in external IdP?

Thank you for Zitadel - great work!

**Comment by hifabienne** [comment]
Hei @nbasov 
Thanks for opening up the discussion.
@stebenz Can you help on this topic?

**Comment by stebenz** [comment]
Hi @nbasov 

To answer your questions:
1. There is no functionality in ZITADEL that checks the state of the user directly if the user is disabled on OKTA and you try to login with OKTA as an external IDP, the login should end in a failure. Or is it still successful? In my opinion, if the user is set to disabled on an external IDP, there should be something that sets the user disabled on ZITADEL afterward, otherwise, it would also be possible through Actions which would have to be improved for this functionality.
2. Currently there is no notification for the user if the state changes, and if that should be through an external IDP it can be trickier as well. The state changes are listed in the console if that's enough, and there is already a notification if you change your password for example, so to send an email could be implemented for the state change.

Hope I could answer everything.
